### DISCUSSION TITLE
Modding Question

### DISCUSSION THREAD

**Discussion by Pelldom** [ekolis/FrEee#219] [Q&A]
Here's a short list of questions from a Modding perspective that I could not find the answers to in the Wiki:

1. "Number of Abilities" and "Number of Techs" fields are no longer needed. How does the game determine the numbers that there are? For example, if there are only 2 abilities, but a mod has the fields for 6 abilities (but the remaining 4 are blank) does that cause a problem?
2. You no longer need to specify indices for abilities and tech requirements. For instance, you can say "Ability Val" instead of "Ability 1 Val 2". Does this apply to Value1 and Value2 for all abilties? So both values would just say "Ability Val"?
3.  parameterized records. It states that these parameters "can be created", but doesn't state where or how. Is there a seperate file that contains these parameters?

**Comment by ekolis** [comment]
1. I don't know what blank abilities would do; you'd have to try that!
2. That's right; both values would be Ability Val.
3. The parameters are explained on this wiki page: https://github.com/ekolis/FrEee/wiki/Modding - you can also set a minimum in addition to a maximum, but the minimum defaults to 1 so you might not need to.

**Reply by Pelldom** [reply]
The syntax and procedures make sense but where do you define the parameters? In the components.txt file? A separate parameters.py file?

**Reply by ekolis** [reply]
The parameters go directly above the first line of the record you're defining them on:

```
Parameter Name := level
Parameter Maximum := 5
Name := Space Yard Facility {level.ToRomanNumeral()}
```
